dev_docs/cxg.md

```diff
@@ -73,38 +70,38 @@ For example:
 }
 ```
 
-### X
+### `X` array
 
-TileDB array, with a single anonymous attribute, shape (n_obs, n_var), containing the count matrix (equivalent to the AnnData `X` array).  MUST have numeric type, and SHOULD be float32.  The TileDB schema defines type and sparsity, and both dense and sparse encoding are supported.
+TileDB array, with a single anonymous attribute, shape `(n_obs, n_var)`, containing the count matrix (equivalent to the AnnData `X` array).  MUST have numeric type, and SHOULD be `float32`.  The TileDB schema defines type and sparsity, and both dense and sparse encoding are supported.
 
-### X_col_shift
+### `X_col_shift` array
 
-Optional TileDB array, used to encode-per column offsets for column-shift sparse encoding.  The TileDB array will have a single anonymous attribute, of the same type as the X array, and shape (n_var,).  
+Optional TileDB array, used to encode per-column offsets for column-shift sparse encoding.  The TileDB array will have a single anonymous attribute, of the same type as the X array, and shape `(n_var,)`.  
 
-If the X array is sparse, and X_col_shift exists, then all values in the i'th column were subtracted by X_col_shift[i].
+If the `X` array is sparse, and the `X_col_shift` array exists, then all values in the `i`'th column were subtracted by `X_col_shift[i]`.
 
-### emb and embedding arrays
+### `emb` group and embedding arrays
 
-A CXG must have a group named `emb`, which will contain all embeddings.  Embeddings are encoded as TileDB arrays, of numeric type and shape (n_obs, >=2).  The arrays SHOULD be coerced to float32, and MUST be a numeric type.  The TileDB array name will be assumed to be the embedding name (conventionally, embedding names in CXG are _not_ prefixed with an `X_` as they are in AnnData).
+A CXG MUST have a group named `emb`, which will contain all embeddings.  Embeddings are encoded as TileDB arrays, of numeric type and shape `(n_obs, >=2)`.  The arrays MUST be a numeric type, and SHOULD be coerced to `float32`. The array values MUST NOT contain any infinite values and MUST contain at least one non-NaN value.  The TileDB array name will be assumed to be the embedding name (conventionally, embedding names in CXG are _not_ prefixed with an `X_` as they are in AnnData).
 
 CXG supports zero or more embeddings. Note that cellxgene currently _requires_ at least one embedding.
 
-### cxg_group_metadata
+### `cxg_group_metadata` group
 
 Required, but empty TileDB array, used to store CXG-wide metadata.  The following fields are defined:
-* __cxg_version__: (required) a semver string identifying the specification version used to encode the CXG.
-* __cxg_properties__: (optional) a dictionary containing dataset wide properties, defined below.
-* __cxg_category_colors__: (optional) a categorical color table, defined below.
+* `__cxg_version__`: (required) a [semantic version](https://semver.org/) identifying the specification version used to encode the CXG.
```

Comment:
these are also malformed due to the addition of the quotes.  The actual key does not include __ prefix or suffix.

Eg, it is `cxg_version`, not `__cxg_version__`

Reply:
I would also add a note to the effect that it must be a semver 2.0 complaint version string.

we were bit by this in an early version of CXG, where I mistakenly hard-wired a malformed version!

Reply:
specified semver 2.0 explicitly.